core/logger/zap.go

```diff
@@ -72,7 +63,7 @@ func (l *zapLogger) Helper(skip int) Logger {
 }
 
 func (l *zapLogger) Name() string {
-	return l.name
+	return l.Desugar().Name()
```

Comment:
They only added it to the normal logger, not the Sugared one, so we still have to implement it ourselves.

Reply:
I'm surprised this doesn't require `l.SugaredLogger.Desugar().Name()`, though it's neat go correctly infers it.